From https://dev.haiku-os.org/ticket/17191

diff --git a/gcc/config/aarch64/aarch64.h b/gcc/config/aarch64/aarch64.h
index a7c2c50e7..ecc3b3e44 100644
--- a/gcc/config/aarch64/aarch64.h
+++ b/gcc/config/aarch64/aarch64.h
@@ -1264,7 +1264,7 @@ extern const char *aarch64_rewrite_mcpu (int argc, const char **argv);
 #define MCPU_TO_MARCH_SPEC_FUNCTIONS \
   { "rewrite_mcpu", aarch64_rewrite_mcpu },
 
-#if defined(__aarch64__)
+#if defined(__aarch64__) && ! defined(__APPLE__)
 extern const char *host_detect_local_cpu (int argc, const char **argv);
 #define HAVE_LOCAL_CPU_DETECT
 # define EXTRA_SPEC_FUNCTIONS						\
diff --git a/gcc/config/host-darwin.c b/gcc/config/host-darwin.c
index b57876715..edec73053 100644
--- a/gcc/config/host-darwin.c
+++ b/gcc/config/host-darwin.c
@@ -23,6 +23,8 @@
 #include "options.h"
 #include "diagnostic-core.h"
 #include "config/host-darwin.h"
+#include "hosthooks.h"
+#include "hosthooks-def.h"
 #include <errno.h>
 
 /* For Darwin (macOS only) platforms, without ASLR (PIE) enabled on the
@@ -181,3 +183,8 @@ darwin_gt_pch_use_address (void *&addr, size_t sz, int fd, size_t off)
 
   return 1;
 }
+
+#if defined(__aarch64__)
+const struct host_hooks host_hooks = HOST_HOOKS_INITIALIZER;
+#endif
+
